src/app/core/components/confirm-gift-dialog/confirm-gift-dialog.component.spec.ts

```diff
@@ -0,0 +1,115 @@
+/* @format */
+import {
+  ComponentFixture,
+  fakeAsync,
+  TestBed,
+  TestModuleMetadata,
+  tick,
+} from '@angular/core/testing';
+import { DialogRef, DIALOG_DATA } from '@root/app/dialog/dialog.module';
+import { SharedModule } from '@shared/shared.module';
+import * as Testing from '@root/test/testbedConfig';
+import { cloneDeep } from 'lodash';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { ApiService } from '@shared/services/api/api.service';
+import { ConfirmGiftDialogComponent } from './confirm-gift-dialog.component';
+
+describe('ConfirmGiftDialogComponent', () => {
+  let component: ConfirmGiftDialogComponent;
+  let fixture: ComponentFixture<ConfirmGiftDialogComponent>;
+  let dialogRef: DialogRef;
+  let dialogData: {
+    email: string;
+    amount: string;
+    message: string;
+    giftResult: Observable<void>;
+  };
+
+  const mockApiService = {
+    billing: {
+      giftStorage: jasmine
+        .createSpy('giftStorage')
+        .and.returnValue(Promise.resolve()),
+    },
+  };
+
+  const mockGiftResult = new BehaviorSubject<boolean>(false);
+
+  beforeEach(async () => {
+    const config: TestModuleMetadata = cloneDeep(Testing.BASE_TEST_CONFIG);
+
+    dialogData = {
+      email: 'test@email.com',
```

Comment:
I usually use "example.com" for things like this, so I don't accidentally use a real email address (even though this won't send any emails). You'll see that in several other places in this codebase, too, so it's good for consistency.